packages/rpc-provider/src/http/index.ts

```diff
@@ -84,8 +84,11 @@ export default class HttpProvider implements ProviderInterface {
    * @description HTTP Provider does not have 'on' emitters. WebSockets should be used instead.
    */
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  public on (type: ProviderInterfaceEmitted, sub: ProviderInterfaceEmitCb): void {
+  public on (type: ProviderInterfaceEmitted, sub: ProviderInterfaceEmitCb): () => void {
     l.error('HTTP Provider does not have \'on\' emitters, use WebSockets instead');
+    return (): void => {
+      // noop
```

Comment:
Is this feasible or do I allow ProviderInterface to also return `void` or  do we split http / ws provider interface type defs?

Reply:
No, it is fine as-is. If we also return void than the called needs to check if it is empty or not - which just complicates things. As it stands, he/she/it can just ignore the return if no action is taken.